Sources/SwiftDocC/Checker/Checkers/TutorialDirectiveMisuseChecker.swift

```diff
@@ -0,0 +1,62 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import Markdown
+
+public struct TutorialDirectiveMisuseChecker: Checker {
+    public var problems = [Problem]()
+
+    private var sourceFile: URL?
+
+    public init(sourceFile: URL?) {
+        self.sourceFile = sourceFile
+    }
+
+    public mutating func visitBlockDirective(_ blockDirective: BlockDirective) {
+        guard let blockRange = blockDirective.range else { return }
+        
+        let diagnostic: Diagnostic
+        let solutions: [Solution]
+
+        switch blockDirective.name {
+        case ImageMedia.directiveName:
```

Comment:
Should we also diagnose `Video` and the other Tutorial-specific directives? @binamaniar added this property recently which should be helpful: https://github.com/apple/swift-docc/pull/154/files#diff-cd174ae1fe7d1732ca971afb59e3ad1575275faa1701aea03c78e8711b505ccaR22

We'd need to subtract `Comment`, `DisplayName`, `DocumentationExtension`, `Metadata`, `Snippet`, `Redirect`, `DeprecationSummary` at least since those can be used in Markdown articles.

Reply:
1. Video is already diagnosed. But only @Image will have solutions. Since Video does not have an equivalent in Markdown form.
2. Oh, I did not know they are supported in Markdown articles. I'll substract them to not produce a diagnost.

Reply:
one way to approach this is have a list of all the directives which are not supported in Markdown and provide a diagnostics for that.

Reply:
One potential enhancement that would be nice here is to have separate lists for documentation extension files (markdown files with a symbol link in the level 1 heading) and articles (other markdown files). `DisplayName`, `DocumentationExtension`, and `DeprecationSummary` are only allowed in documentation extension files.